### PR TITLE
Calculate issue age

### PR DESCRIPTION
**Pull request by Lehcar**
cc/ https://github.com/sbv-world-health-org-metrics/sbv-world-health-org-metrics/issues/72

### CODE REVIEW

File: backend/business/core/collectors/github/fetchers/repos_info_fetcher.go [hunk 1]
```diff
@@ -54,6 +56,11 @@ type repoInfo struct {
 		}
 		OpenIssues struct {
 			TotalCount githubv4.Int
+			Nodes      []struct {
+				Issue struct {
+					CreatedAt githubv4.DateTime
+				}
+			}
```
Comment (ajhenry) [inline backend/business/core/collectors/github/fetchers/repos_info_fetcher.go:63]:
```suggestion
			Nodes      []struct {
				CreatedAt githubv4.DateTime
			}
```

Comment (ajhenry) [reply backend/business/core/collectors/github/fetchers/repos_info_fetcher.go:63]: This was just one level of indirection away from being able to run 🏃 

`Issues` isn't a field on `Nodes` so we can't access it. But once we remove it then we're good to go

File: backend/business/core/collectors/github/fetchers/repos_info_fetcher.go [hunk 2]
```diff
@@ -124,6 +131,7 @@ func (f *ReposInfoFetcher) buildResult(data []repoInfo) *map[string]RepoInfoResu
 			ForksCount:              int(edge.Node.Forks.TotalCount),
 			IssuesCount:             int(edge.Node.Issues.TotalCount),
 			OpenIssuesCount:         int(edge.Node.OpenIssues.TotalCount),
+			OpenIssuesCreatedAt:     time.Time(edge.Node.OpenIssues.Nodes[len(edge.Node.OpenIssues.Nodes)-1].Issue.CreatedAt.Time),
```
Comment (ajhenry) [inline backend/business/core/collectors/github/fetchers/repos_info_fetcher.go:134]: This present an interesting problem to our set in that we should probably calculate the median/average for this timestamp but to do that effectively we'll need to rewrite the fetcher above to paginate results for us. @ipc103 and I came up with this function as a starting point


```go
func (f *ReposInfoFetcher) buildResult(data []repoInfo) *map[string]RepoInfoResult {
	holder := make(map[string]RepoInfoResult)

	// find the median age of open issues
	
	for _, edge := range data {
		var medianCreatedTime time.Time
		medianIssueIndex := len(edge.Node.OpenIssues.Nodes) / 2
		if medianIssueIndex > 0 {
			medianCreatedTime = time.Time(edge.Node.OpenIssues.Nodes[medianIssueIndex].CreatedAt.Time)
		}


		holder[string(edge.Node.NameWithOwner)] = RepoInfoResult{
			RepoName:                string(edge.Node.NameWithOwner),
			CollaboratorsCount:      int(edge.Node.Collaborators.TotalCount),
			ProjectsCount:           int(edge.Node.Projects.TotalCount + edge.Node.ProjectsV2.TotalCount),
			DiscussionsCount:        int(edge.Node.Discussions.TotalCount),
			ForksCount:              int(edge.Node.Forks.TotalCount),
			IssuesCount:             int(edge.Node.Issues.TotalCount),
			OpenIssuesCount:         int(edge.Node.OpenIssues.TotalCount),
			OpenIssuesCreatedAt:     medianCreatedTime,
			ClosedIssuesCount:       int(edge.Node.ClosedIssues.TotalCount),
			OpenPullRequestsCount:   int(edge.Node.OpenPullRequests.TotalCount),
			MergedPullRequestsCount: int(edge.Node.MergedPullRequests.TotalCount),
			LicenseName:             string(edge.Node.LicenseInfo.Name),
			WatchersCount:           int(edge.Node.Watchers.TotalCount),
		}
	}

	return &holder
}
```

**Comment by ipc103** [comment]
Closing in favor of #93

@Lehcar We found that this was much easier to implement with a pipeline-based system using JS/TS. I'm going to write some more detailed documentation about how this works and will share shortly. Sorry to close this out without more discussion on the PR, but I think this will make things much easier moving forward.

**Comment by Lehcar** [comment]
> Closing in favor of #93
> 
> @Lehcar We found that this was much easier to implement with a pipeline-based system using JS/TS. I'm going to write some more detailed documentation about how this works and will share shortly. Sorry to close this out without more discussion on the PR, but I think this will make things much easier moving forward.
>
Awesome! Thanks for doing the JS/TS conversion! Do you know if we can close out https://github.com/sbv-world-health-org-metrics/sbv-world-health-org-metrics/issues/72 now?

**Pull request closed without merging**